File: Arrays_and_Strings/Two_Sum_p.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int, int>> vals;
        for(int i=0; i<nums.size();  i++) {
            vals.push_back({nums[i], i});
        }
        sort(vals.begin(), vals.end());
        int i=0, j=vals.size()-1;
        while(i<j) {
            if(vals[i].first + vals[j].first < target) i++;
            else if(vals[i].first + vals[j].first > target) j--;
            else return vector<int> {vals[i].second, vals[j].second};
        }
        return vector<int>{};
    }
};

File: Arrays_Strings_Linked_Lists/Array_Basics/Two_Sum_p.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            nums[i] = (nums[i], i)
        nums.sort()
        i = 0
        j = len(nums)-1
        while i<j:
            if nums[i][0] + nums[j][0] > target:
                j -= 1
            elif nums[i][0] + nums[j][0] < target:
                i += 1
            else:
                return [nums[i][1], nums[j][1]]
        return []